ui/page/governance/proposal_details_page.go

```diff
@@ -129,26 +131,32 @@ func (pg *ProposalDetails) HandleUserInteractions() {
 }
 
 func (pg *ProposalDetails) listenForSyncNotifications() {
+	if pg.ProposalNotificationListener == nil {
+		pg.ProposalNotificationListener = listeners.NewProposalNotificationListener()
+	}
+	err := pg.WL.MultiWallet.Politeia.AddNotificationListener(pg.ProposalNotificationListener, ProposalDetailsPageID)
+	if err != nil {
+		log.Errorf("Error adding Politeia notification listener: %+v", err)
+		return
+	}
 
 	go func() {
 		for {
-			var notification interface{}
-
 			select {
-			case notification = <-pg.Receiver.NotificationsUpdate:
-			case <-pg.ctx.Done():
-				return
-			}
-
-			switch n := notification.(type) {
-			case wallet.Proposal:
-				if n.ProposalStatus == wallet.Synced {
+			case notification := <-pg.ProposalNotifChan:
+				switch notification.ProposalStatus {
+				case wallet.Synced:
 					proposal, err := pg.WL.MultiWallet.Politeia.GetProposalRaw(pg.proposal.Token)
 					if err == nil {
 						pg.proposal = proposal
 						pg.RefreshWindow()
 					}
 				}
+			case <-pg.ctx.Done():
+				pg.WL.MultiWallet.Politeia.RemoveNotificationListener(ProposalDetailsPageID)
+				close(pg.ProposalNotifChan)
+
```

Comment:
I think you should also set `pg.ProposalNotificationListener` to nil so that if this method is recalled, it'll be re-initialized instead of trying to use a closed channel.